lib/src/native/realm_core.dart

```diff
@@ -850,6 +850,36 @@ class _RealmCore {
     final realm_app = _realmLib.invokeGetPointer(() => _realmLib.realm_app_get(appConfig._pointer, syncClientConfig._pointer));
     return AppHandle._(realm_app);
   }
+
+  static void void_completion_callback(Pointer<Void> userdata, Pointer<realm_app_error> error) {
+    final Completer<void>? completer = userdata.toObject();
+    if (completer == null) {
+      return;
+    }
+
+    if (error != nullptr) {
+      final message = error.ref.message.cast<Utf8>().toDartString();
+      completer.completeError(RealmException(message));
+      return;
+    }
+
+    completer.complete();
+  }
+
+  Future<void> appEmailPasswordRegisterUser(Application application, String email, String password) {
+    final completer = Completer<void>();
+    using((arena) {
+      _realmLib.invokeGetBool(() => _realmLib.realm_app_email_password_provider_client_register_email(
+            application.handle._pointer,
+            email.toUtf8Ptr(arena),
+            password.toRealmString(arena).ref,
+            Pointer.fromFunction(void_completion_callback),
+            completer.toGCHandle(),
```

Comment:
I would suggest using a hard handle here, as there are no guarantee that the client will ever await the returned future, and hence the local completer may become garbage before the void_completion_callback calls `userdata.toObject()`

Reply:
yes I will be fixing all these calls to use "hard handles"

Reply:
btw, problem is that the future does not hold of the completer at all. So this is definitelly crashing.

Reply:
this should be fixed now